src/util/taproot.rs

```diff
@@ -790,16 +821,15 @@ impl LeafVersion {
         match version {
             TAPROOT_LEAF_TAPSCRIPT => Ok(LeafVersion::TapScript),
             TAPROOT_ANNEX_PREFIX => Err(TaprootError::InvalidTaprootLeafVersion(TAPROOT_ANNEX_PREFIX)),
-            odd if odd & TAPROOT_LEAF_MASK != odd => Err(TaprootError::InvalidTaprootLeafVersion(odd)),
-            future => Ok(LeafVersion::Future(future)),
+            future => FutureLeafVersion::from_consensus(future).map(LeafVersion::Future),
```

Comment:
In ef8a3a839e5f0faccf9f6299b295309f2eab3727:

We should refuse to parse odd leaf versions.

Reply:
Thry are refused at the level of `FutureLeafVersion`

Reply:
Derp, you're right, thanks.